bundles/binding/org.openhab.binding.insteonplm/src/main/java/org/openhab/binding/insteonplm/InsteonPLMActiveBinding.java

```diff
@@ -299,6 +299,19 @@ public void updated(Dictionary<String, ?> config) throws ConfigurationException
 		if (!m_hasInitialItemConfig) triggerBindingChangedCalls();
 		return;
 	}
+	/**
+	 * Method to find a device by address
+	 * @param aAddr the insteon address to search for
+	 * @return reference to the device, or null if not found
+	 */
+	public InsteonDevice getDevice(InsteonAddress aAddr) {
+		InsteonDevice dev = null;
+		synchronized (m_devices) {
```

Comment:
rather than synchronizing each access to m_devices i would suggest to use ConcurrentHashMap (see also http://stackoverflow.com/questions/510632/whats-the-difference-between-concurrenthashmap-and-collections-synchronizedmap)